### PR TITLE
Settings Page

### PR DESCRIPTION
**Pull request by Dobos23**
Added a functional Settings Page enabling users to change usernames and passwords.

### CODE REVIEW

File: backend/src/controllers/UserController.ts [hunk 1]
```diff
@@ -64,6 +64,21 @@ class UserController {
...
+
+  public async updateUser(req: Request, res: Response) {
+    try {
+      const { userId, username, password } = req.body;
+      const success = await UserService.updateUser(
+        userId,
+        username,
+        password
+      );
+      res.status(200).json({success});
```
Comment (pastelnata) [inline backend/src/controllers/UserController.ts:76]: dont forget to generate a new token after updating the user

**Review by pastelnata — CHANGES_REQUESTED** [review]

**Comment by alikrist** [comment]
branch has conflicts in front-end routes

**Comment by Dobos23** [comment]
conflicts solved, the rest is handled by Hubert

**Comment by pastelnata** [comment]
image

**Comment by pastelnata** [comment]
I solved but but now i cannot merge cause i pushed changes lol

**Review by sduhubert — APPROVED** [review]
slauy

**Pull request merged**